api/handler/rpc/rpc.go

```diff
@@ -300,13 +300,17 @@ func requestPayload(r *http.Request) ([]byte, error) {
 
 	// allocate maximum
 	matches := make(map[string]interface{}, len(md))
+	bodydst := ""
 
 	// get fields from url path
 	for k, v := range md {
 		// filter own keys
 		if strings.HasPrefix(k, "x-api-field-") {
 			matches[strings.TrimPrefix(k, "x-api-field-")] = v
 			delete(md, k)
+		} else if k == "x-api-body" {
```

Comment:
What sets these metadata values?

Reply:
they set by router (i'm plan to add it also for registry router, now it filled only in static router)

Reply:
Should the router not just set some informational struct in context?

Reply:
Should it not just maybe pass in the endpoint in context?

Reply:
i don't have any objections about using router context, this code mostly based on decision in slack when chat with you. We want to have something that working and iterate, request context used because it already present. if we use dedicated router context - this save some memory to avoid growing map

Reply:
Lets merge this first iteration then move to using a context passed through